app/(modals)/settings.tsx

```diff
@@ -1,12 +1,50 @@
-import {View, Text} from 'react-native';
-import React from 'react';
+import {View, Text, TouchableOpacity, StyleSheet} from 'react-native';
+import React, {useState}  from 'react';
 
 const Settings = () => {
+  const [generatedKey, setGeneratedKey] = useState<string|null>(null);
+
+  const handleGenerateKey = () => {
+    const newKey = Math.random().toString(36).substring(2, 10);
```

Comment:
Quick note: I think we could use the { createHash } from crypto library in the future instead of Math.random due to security reasons.